components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -248,6 +252,26 @@ protected void processAuthenticationResponse(HttpServletRequest request,
         }
     }
 
+    private void handleErrorResponse(HttpServletRequest request, HttpServletResponse response,
+                                     AuthenticationContext context)
+            throws InvalidCredentialsException {
+        StringBuilder errorMessage = new StringBuilder();
+        if (request.getParameter(FacebookAuthenticatorConstants.OAUTH2_PARAM_ERROR) != null) {
+            errorMessage.append("error_code: ")
+                    .append(request.getParameter(FacebookAuthenticatorConstants.OAUTH2_PARAM_ERROR_CODE))
+                    .append(", error: ")
+                    .append(request.getParameter(FacebookAuthenticatorConstants.OAUTH2_PARAM_ERROR))
+                    .append(", error_description: ")
+                    .append(request.getParameter(FacebookAuthenticatorConstants.OAUTH2_PARAM_ERROR_DESCRIPTION))
+                    .append(", error_reason: ")
+                    .append(request.getParameter(FacebookAuthenticatorConstants.OAUTH2_PARAM_ERROR_REASON));
+            if(log.isDebugEnabled()) {
+                log.debug("Failed to authenticate via Facebook. " + errorMessage.toString());
+            }
+            throw new InvalidCredentialsException(errorMessage.toString());
```

Comment:
Throw InvalidCredentialsException instead of AuthenticationFailedException, to avoid the error log printing (refer [1]).
[1] https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java#L549